packages/common/src/types/pivot.ts

```diff
@@ -6,37 +6,58 @@ export type PivotConfig = {
     metricsAsRows: boolean;
     columnOrder?: string[];
     hiddenMetricFieldIds?: string[];
+
+    columnTotals?: boolean;
+    rowTotals?: boolean;
 };
 
-type HeaderOrIndexType =
+type PTField =
     | { type: FieldType.METRIC; fieldId?: undefined }
     | { type: FieldType.DIMENSION; fieldId: string };
 
-export type PivotHeaderType = HeaderOrIndexType;
-export type PivotIndexType = HeaderOrIndexType;
+export type PTFieldLabel = {
+    type: 'label';
+    fieldId: string;
+    value?: undefined;
+};
+
+export type PTTotalLabel = {
+    type: 'total';
+};
 
-export type PivotValue =
-    | { type: 'label'; fieldId: string; value?: undefined }
-    | { type: 'value'; fieldId: string; value: ResultValue };
+export type PTValue = {
+    type: 'value';
+    fieldId: string;
+    value: ResultValue;
+};
 
-export type PivotTitleValue = PivotValue & {
+export type PTLabel = {
     titleDirection: 'index' | 'header';
 };
 
-export type PivotData = {
-    titleFields: (PivotTitleValue | null)[][];
+export type PTTotalValue = Pick<ResultValue, 'raw'>;
 
-    headerValueTypes: PivotHeaderType[];
-    headerValues: PivotValue[][];
+export type PTTitleValue = PTFieldLabel & PTLabel;
```

Comment:
Why does this include `type: 'label'` or `value?: undefined`  if we never check those fields?

This type should be:

```
export type PivotTableTitleValue = {
  fieldId: string;
  titleDirection: 'index' | 'header';
}
```